### PR TITLE
Revise and consolidate osx-arm64 commits

### PR DESCRIPTION
**Pull request by ldennington**
This change revises the commits from #636 by:

1. Applying existing fixups.
2. Consolidating the series of commits to add `osx-arm64` support into two commits. The first commit lays the ground work for the changes by updating certain dependencies. The second commit implements `osx-arm64` support.

### CODE REVIEW

File: src/osx/Installer.Mac/distribution.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <installer-gui-script minSpecVersion="1">
     <title>Git Credential Manager</title>
     <background file="background.png" mime-type="image/png" alignment="bottomleft" scaling="tofit" />
-    <options customize="never" />
+    <options customize="never" hostArchitectures="x86_64,arm64" />
```
Comment (mjcheetham) [inline src/osx/Installer.Mac/distribution.xml:5]: I realise this was also in the first PR, but is correct?

Comment (ldennington) [reply src/osx/Installer.Mac/distribution.xml:5]: Mmmm no. Looks like the separate distribution files may have been dropped in the midst of a rebase. I will bring them back.

**Review by mjcheetham — APPROVED** [review]
🎉

File: src/osx/Installer.Mac/distribution.xml (hunk 1 above)
Comment (Rhondad3) [reply src/osx/Installer.Mac/distribution.xml]: No

**Pull request merged**